tests/integration/test_webhook.py

```diff
@@ -0,0 +1,137 @@
+            )
+
+        # get all running executions
+        running_executions = sfn_client.list_executions(
+            stateMachineArn=step_function_arn, statusFilter="RUNNING"
+        )["executions"]
+
+        if len(running_executions) == 1:
+            execution = running_executions[0]
+            execution_results = sfn_client.get_execution_history(
+                executionArn=execution["executionArn"]
+            )
+            return json.loads(
+                execution_results["events"][0]["executionStartedEventDetails"]["input"]
+            )
+
+        if len(running_executions) > 1:
+            raise Exception(
+                f"Found {len(running_executions)} RUNNING executions, cannot reliably "
+                f"determine if invoked by this webhook event."
+            )
+
+        time.sleep(0.5)
+
+
+@pytest.mark.integration()
+@pytest.mark.usefixtures("_set_integration_test_environ")
+def test_integration_webhook_handles_get_request_success():
+    """Test that deployed lambda can receive GET requsts"""
+    challenge_phrase = "challenge-accepted"
+    response_text = send_get_to_lambda_function_url(challenge_phrase)
+    assert response_text == challenge_phrase
+
+
+@pytest.mark.integration()
+@pytest.mark.usefixtures("_set_integration_test_environ")
+def test_integration_webhook_handles_post_request_but_no_job_type_match():
+    """Test that deployed lambda can receive POST requests and skips unknown job type"""
+    payload = {"action": "JOB_END", "job_instance": {"name": "Unhandled Job Name Here"}}
+    response_text = send_post_to_lambda_function_url(payload)
+    assert (
+        response_text == "Webhook POST request received and validated in dev env for "
+        "job 'Unhandled Job Name Here', no action taken."
+    )
+
+
+@pytest.mark.integration()
+@pytest.mark.usefixtures("_set_integration_test_environ")
+@pytest.mark.usefixtures("_integration_tests_s3_fixtures")
+def test_integration_webhook_handles_pod_step_function_trigger(
+    sample_pod_export_job_end_webhook_post_body,
+):
+    """Test deployed lambda handles PPOD job type webhooks and invokes step function.
+
+    Able to assert specific values in StepFunction execution results based on fixtures.
+    """
+    response_text = send_post_to_lambda_function_url(
+        sample_pod_export_job_end_webhook_post_body
+    )
+
+    # assert lambda response that StepFunction invoked
+    assert (
+        response_text == "Webhook POST request received and validated, PPOD pipeline "
+        "initiated."
+    )
+
+    # assert StepFunction running and with expected payload
+    step_function_arn = os.getenv("PPOD_STATE_MACHINE_ARN")
+    step_function_input_json = get_step_function_invocation_results(step_function_arn)
+    assert (
+        step_function_input_json["filename-prefix"]
+        == "exlibris/pod/POD_ALMA_EXPORT_20230815"
+    )
+
+
+@pytest.mark.integration()
+@pytest.mark.usefixtures("_set_integration_test_environ")
```

Comment:
Is there a difference in functionality between using the `@pytest.mark.usefixtures` decorator vs. passing the fixture into the test definition (i.e., passing it inside the parentheses)?

I don't have a preference but was curious!

Reply:
Ah, right! I think I was notified of this syntax with Ruff previously. Thanks for the heads up!